server/vmap/trillian_map_server.go

```diff
@@ -37,12 +34,12 @@ func (t *TrillianMapServer) getHasherForMap(mapID int64) (merkle.MapHasher, erro
 // GetLeaves implements the GetLeaves RPC method.
 func (t *TrillianMapServer) GetLeaves(ctx context.Context, req *trillian.GetMapLeavesRequest) (resp *trillian.GetMapLeavesResponse, err error) {
 	ctx = util.NewMapContext(ctx, req.MapId)
-	s, err := t.registry.GetMapStorage(req.MapId)
+	s, err := t.registry.GetMapStorage()
```

Comment:
I'd rather have only the registry as an attribute and keep calling GetMapStorage() for each RPC. We can return the same pointer from GetMapStorage if we want (although creating a new Storage object is fairly cheap now with the changes in this PR). I think this is better solved there (at the registry) than here.